simd/src/Kokkos_Simd_Scalar.hpp

```diff
@@ -0,0 +1,299 @@
+#ifndef KOKKOS_SIMD_SCALAR_HPP
+#define KOKKOS_SIMD_SCALAR_HPP
+
+#include <type_traits>
+#include <limits.h>
+
+#include <Kokkos_Simd_Common.hpp>
+
+namespace Kokkos {
+namespace Experimental {
+
+namespace simd_abi {
+
+class scalar {};
+
+}  // namespace simd_abi
+
+template <class T>
+class simd_mask<T, simd_abi::scalar> {
+  bool m_value;
+
+ public:
+  using value_type                      = bool;
+  using simd_type                       = simd<T, simd_abi::scalar>;
+  using abi_type                        = simd_abi::scalar;
+  using reference                       = value_type&;
+  KOKKOS_DEFAULTED_FUNCTION simd_mask() = default;
+  KOKKOS_FORCEINLINE_FUNCTION static constexpr std::size_t size() { return 1; }
```

Comment:
```suggestion
  KOKKOS_FUNCTION static constexpr std::size_t size() { return 1; }
```
Same comment for all member functions defined in the body of the class.

Reply:
As I commented above, I feel that FORCEINLINE is useful for these methods, but I'm willing to profile and switch to regular `KOKKOS_FUNCTION`